src/main/java/com/evolvedbinary/tulip/XPath10Lexer.java

```diff
@@ -15,9 +15,44 @@
 package com.evolvedbinary.tulip;
 
 import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
 
 public class XPath10Lexer extends AbstractLexer {
 
+    public static Trie trie = new Trie();
+
+    static {
+        // Insert all axis names
+        for (String axis : new String[]{
+                "ancestor", "ancestor-or-self", "attribute", "child",
+                "descendant", "descendant-or-self", "following",
+                "following-sibling", "namespace", "parent",
+                "preceding", "preceding-sibling", "self"
+        }) {
+            trie.insert(axis, false);
+        }
+
+        // Insert all function names
+        for (String function : new String[]{
+                // Node set functions
```

Comment:
We should not need to know the function names at all at the Lexing stage, instead it should be enough to just recognise function names as Identifiers.